From 03ad17e6281ebfd25a9747cea468be428f5af990 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 7 Oct 2022 23:52:15 +0530
Subject: [PATCH] hostapd: Frequency check during channel switch across BW

Add the check for verifying the frequency for different BW
received during channel switch.

Check makes sure that the channel frequency is appropriate
and lies within the center frequency.

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 hostapd/ctrl_iface.c | 145 +++++++++++++++++++++++--------------------
 1 file changed, 76 insertions(+), 69 deletions(-)

--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -2764,19 +2764,23 @@ static int hostapd_ctrl_check_freq_param
 		switch (params->sec_channel_offset) {
 		case 1:
 			if (params->freq + 150 != params->center_freq1 &&
-			    params->freq + 70 != params->center_freq1 &&
-			    params->freq + 30 != params->center_freq1 &&
-			    params->freq - 10 != params->center_freq1 &&
-			    params->freq - 50 != params->center_freq1 &&
+			    params->freq + 110 != params->center_freq1 &&
+			    params->freq + 70  != params->center_freq1 &&
+			    params->freq + 30  != params->center_freq1 &&
+			    params->freq - 10  != params->center_freq1 &&
+			    params->freq - 50  != params->center_freq1 &&
+			    params->freq - 90  != params->center_freq1 &&
 			    params->freq - 130 != params->center_freq1)
 				return -1;
 			break;
 		case -1:
 			if (params->freq + 130 != params->center_freq1 &&
-			    params->freq + 50 != params->center_freq1 &&
-			    params->freq + 10 != params->center_freq1 &&
-			    params->freq - 30 != params->center_freq1 &&
-			    params->freq - 70 != params->center_freq1 &&
+			    params->freq + 90  != params->center_freq1 &&
+			    params->freq + 50  != params->center_freq1 &&
+			    params->freq + 10  != params->center_freq1 &&
+			    params->freq - 30  != params->center_freq1 &&
+			    params->freq - 70  != params->center_freq1 &&
+			    params->freq - 110 != params->center_freq1 &&
 			    params->freq - 150 != params->center_freq1)
 				return -1;
 			break;
